InnerEye/ML/SSL/encoders.py

```diff
@@ -80,9 +80,8 @@ def get_encoder_output_dim(
             SSLOnlineEvaluatorInnerEye,
         )
 
-        dataloader = dm.train_dataloader()
-        dataloader = dataloader[SSLDataModuleType.LINEAR_HEAD] if isinstance(dataloader, dict) else dataloader  # type: ignore
-        batch = iter(dataloader).next()  # type: ignore
+        batch = next(iter(dm.train_dataloader()))
+        batch = batch[SSLDataModuleType.LINEAR_HEAD] if isinstance(batch, dict) else batch  # type: ignore
```

Comment:
Question rather than feedback: does it slow it down to index every batch rather than indexing the dataloader object as before?

Reply:
@ant0nsc - this is the comment we were referring to

Reply:
1) This is only working on the first element of the data loader, so I would not worry too much about it
2) we can't index in the same way as before anymore because train_dataloader no longer returns a dictionary.